File: src/web/src/models/registration.ts
export default class Registration {
    creationDate: string;
    isPublic: boolean;
    nickname: string;
    ownerID: number;
    teamID: number;

    constructor (creationDate: string, isPublic: boolean, nickname: string, ownerID: number, teamID: number){
        this.creationDate = creationDate ?? "";
        this.isPublic = isPublic;
        this.nickname = nickname ?? "";
        this.ownerID = ownerID ?? -1;
        this.teamID = teamID ?? -1;
    }
}

File: src/combined/Models/Entities/Registration.cs
namespace Api.Models.Entities;

public class Registration
{
    public long ID { get; init; }

    public DateTime CreationDate { get; init; }

    public bool IsPublic { get; init; }

    public string? Nickname { get; init; }

    public long OwnerID { get; init; }

    public long TeamID { get; init; }

    public IEnumerable<PersonRegistration> PersonRegistrations { get; init; }

}
